### ISSUE TITLE
Must statement not applied if it is in a grouping's grouping

### ISSUE THREAD

**Issue by szabolcsszekely** [mbj4668/pyang#208]
When I have a must statement (statement-1) inside a grouping which is used by another grouping with a must statement (statement-2) which is used by the module then statement-1 is not applied during XML validation only statement-2.
Example:
```
$ cat main.yang 
module main {
    yang-version 1;
    namespace "http://example.com/main";
    prefix main;
    grouping inside-group {
        container must-test {
            must "must-test-1 = 1" {
                error-message "must inside grouping's grouping";
            }
            leaf must-test-1 {
                type uint8;
            }
            leaf must-test-2 {
                type uint8;
            }
        }
    }
    grouping group {
        container group {
            uses inside-group;
            must "must-test/must-test-2 = 1" {
                error-message "must inside grouping";
            }
        }
    }
    uses group;
}
$ cat test.xml 
<?xml version='1.0' encoding='UTF-8'?>
<config xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
    <group xmlns="http://example.com/main">
        <must-test>
            <must-test-1>2</must-test-1>
            <must-test-2>2</must-test-2>
        </must-test>
    </group>
</config>
$ yang2dsdl -t config -v test.xml main.yang 
== Generating RELAX NG schema './main-config.rng'
Done.

== Generating Schematron schema './main-config.sch'
Done.

== Generating DSRL schema './main-config.dsrl'
Done.

== Validating grammar and datatypes ...
test.xml validates

== Adding default values... done.

== Validating semantic constraints ...
--- Failed assert at "/nc:config/main:group":
    must inside grouping
```
If I put a refine must statement (statement-3) after the uses inside-group statement then statement-1 is applied but statement-3 is duplicated in the dsdl file.
Example (the test.xml is the same):
```
$ cat main.yang
module main {
    yang-version 1;
    namespace "http://example.com/main";
    prefix main;
    grouping inside-group {
        container must-test {
            must "must-test-1 = 1" {
                error-message "must inside grouping's grouping";
            }
            leaf must-test-1 {
                type uint8;
            }
            leaf must-test-2 {
                type uint8;
            }
        }
    }
    grouping group {
        container group {
            uses inside-group {
                refine must-test {
                    must "must-test-1 = 1" {
                        error-message "must inside grouping refine";
                    }
                }
            }
            must "must-test/must-test-2 = 1" {
                error-message "must inside grouping";
            }
        }
    }
    uses group;
}
$ yang2dsdl -t config -v test.xml main.yang 
== Generating RELAX NG schema './main-config.rng'
Done.

== Generating Schematron schema './main-config.sch'
Done.

== Generating DSRL schema './main-config.dsrl'
Done.

== Validating grammar and datatypes ...
test.xml validates

== Adding default values... done.

== Validating semantic constraints ...
--- Failed assert at "/nc:config/main:group":
    must inside grouping
--- Failed assert at "/nc:config/main:group/main:must-test":
    must inside grouping refine
--- Failed assert at "/nc:config/main:group/main:must-test":
    must inside grouping's grouping
--- Failed assert at "/nc:config/main:group/main:must-test":
    must inside grouping refine
```
I think must statements should work independent of in which grouping it is defined or I missed something from the RFC.

**Comment by szabolcsszekely** [comment]
@llhotka do you have any update on this issue?

**Comment by szabolcsszekely** [comment]
Thanks for the correction.
Did you check the duplicated error-message in case of refine? (the "must inside grouping refine" error message)

**Comment by llhotka** [comment]
I think the current behaviour is correct. The *refine* statement adds a new *must* constrains, the one defined in the `inside-group` grouping isn't replaced – both constraints apply even though their expression is the same.

**Comment by llhotka** [comment]
Sorry, I missed your point, now I see what you mean - I'll fix it.

**Comment by llhotka** [comment]
This was caused by pull request #189 that was merged in 0430aa5. I reverted the changes, and the duplication in your example is gone.

**Comment by szabolcsszekely** [comment]
Thanks for the correction, there are no duplicate error-messages.
